bats/tests/helpers/commands.bash

```diff
@@ -27,3 +27,7 @@ fi
 
 RDSHELL="$RDCTL shell"
 RDSUDO="$RDSHELL sudo"
+
+#After running factory reset, we need to call rdctl from resources as it was removed from the path
+RESOURCES=$(set_resources)
+RDCTL_resources="$RESOURCES/bin/rdctl"
```

Comment:
I think I would rather define a `rdctl` function (the same way we define the `limactl` function):

```suggestion
rdctl() {
    "$(resources_dir)/bin/$RDCTL" "$@"
}
```

And then use this function everywhere instead of `$RDCTL`. This should probably be done later in a separate PR.

Reply:
I believe we should keep the variable $RDCTL from $PATH because this would be exactly as it is in the use case. Once the user runs rdctl factory-reset from the CLI/UI and the path management is set to automatic, rdctl is got from the path, not from resources. We could create this function to validate that when the path-management is set to manual, get rdctl from resources. Otherwise, get rdctl from $PATH. What do you think?

Reply:
I'm keeping `$RDCTL` for now, but I'm not sure why we wouldn't want to use `rdctl` everywhere.

Reply:
I'm now convinced that we should get rid of all the `$RDCTL` and like commands and replace them with functions invoking them with the full path from the resources directory:

The current mechanism relies on BATS being run from a shell that has `~/.rd/bin` already on the `PATH`. If it doesn't, then invoking `rdctl start --path-management-strategy rcfiles` doesn't help because it only updates `~/.bash_profile`, but not the current process environment.[^1]

So invoking the bundled utilities with the proper full path will make the tests more robust.

I will make this change in a future PR.

[^1]: This is assuming that `rdctl start --path-management-strategy rcfiles` works, which it currently doesn't. I've temporarily added code to this PR to work around this bug.